lib/fluent/plugin/filter_masking.rb

```diff
@@ -11,18 +11,37 @@ def strToHash(str)
         eval(str)
       end
 
+      def myDig(input, path)
+        curr = input
+        for segment in path do
+          if curr != nil && curr.respond_to?(:dig)
```

Comment:
since the `dig` method isn't called here but used for checking if `curr` is of object type hash, i recommend changing to

 ```suggestion
           if curr != nil && curr.is_a?(Hash)
```

it is more understandable that way IMO